feat: new documentation about KMP use cases (#461)
JetBrains/kotlin-multiplatform-dev-docs: topics/overview/use-cases-examples.md

**Case studies:** [Duolingo](https://youtu.be/RJtiFt5pbfs?si=mFpiN9SNs8m-jpFL), [Quizlet](https://quizlet.com/blog/shared-code-kotlin-multiplatform), [Chalk](https://kotlinlang.org/lp/multiplatform/case-studies/chalk/?_gl=1*1wxmdrv*_gcl_au*MTE5NzY3MzgyLjE3NDk3MDk0NjI.*FPAU*MTE5NzY3MzgyLjE3NDk3MDk0NjI.*_ga*MTM4NjAyOTM0NS4xNzM2ODUwMzA5*_ga_9J976DJZ68*czE3NTEwMjI5ODAkbzYwJGcxJHQxNzUxMDIzMTU2JGo1OCRsMCRoMA..), [Memrise](https://engineering.memrise.com/kotlin-multiplatform-memrise-3764b5a4a0db), Physics Wallah

### E-commerce

Building cross-platform shopping experiences means balancing shared business logic with native features like payments, camera access, and maps. Kotlin Multiplatform with Compose Multiplatform enables teams to share both business logic and UIs across platforms, while still using platform-specific components where needed. This hybrid approach ensures faster development, a consistent user experience, and the flexibility to integrate critical native features.

> **Case studies:** Balary Market, Markaz

Reviewer: For all "empty" items, I can add links to the corresponding descriptions on our case studies page. What do you think?

Author: If you could do it, I would appreciate it a lot! I tried to do it myself, but failed to find the correct link leading to specific descriptions.

> 
{style="note"}

From integrating into existing apps and sharing app logic to building new cross-platform applications, 
[Kotlin Multiplatform](https://www.jetbrains.com/kotlin-multiplatform/) has become the technology of choice for many companies. These teams are capitalizing on the 
advantages that KMP provides to roll out their products faster and reduce development costs.

A growing number of businesses are also adopting [Compose Multiplatform](https://www.jetbrains.com/compose-multiplatform/), a declarative UI framework powered by 
Kotlin Multiplatform and Google’s Jetpack Compose. With the [stable release for iOS](https://blog.jetbrains.com/kotlin/2025/05/compose-multiplatform-1-8-0-released-compose-multiplatform-for-ios-is-stable-and-production-ready/), Compose Multiplatform completes the picture, 
making KMP a complete solution for cross-platform mobile development.

As adoption grows, this article examines how Kotlin Multiplatform is used in production across different industries and team structures.

## Kotlin Multiplatform use cases by the type of business and team

> Here are several ways different teams are leveraging Kotlin Multiplatform to meet various project needs:

Reviewer: ```suggestion
Here are several ways different teams are applying Kotlin Multiplatform to meet various project needs:
```

### Startups starting a new greenfield project

Startups often operate with limited resources and tight deadlines. To maximize development efficiency and cost-effectiveness, 
they benefit from targeting multiple platforms using a shared codebase — especially in early-stage products or MVPs, 
where time-to-market is critical.

For companies that want to share both logic and UI, Kotlin Multiplatform with Compose Multiplatform offers an ideal solution. 
You can start with a shared UI, enabling quick prototyping. You can even mix and match native UIs with shared UIs. 
This makes KMP with Compose Multiplatform an ideal choice for greenfield projects, helping startups balance speed, 
flexibility, and a high-quality native experience.

**Case studies:**

* [Instabee](https://www.youtube.com/watch?v=YsQ-2lQYQ8M) migrated its Android application logic and UI to Kotlin Multiplatform with Compose Multiplatform. 

>   By leveraging the Android codebase effectively, the company was able to release its iOS application in a short period of time.

Reviewer: ```suggestion
  By using the Android codebase effectively, the company was able to release its iOS application in a short period of time.
```

* [Respawn Pro](https://youtu.be/LB5a2FRrT94?si=vgcJI-XoCrWree3u) develops a habit-tracking and productivity app. Its iOS app is built with Compose Multiplatform, 
  sharing 96% of its code with Android.

> If you’re choosing between [Kotlin Multiplatform and Flutter](https://www.jetbrains.com/help/kotlin-multiplatform-dev/kotlin-multiplatform-flutter.html), 
> don’t miss our overview of both technologies.
> 
{style="tip"}

### Small and medium-sized businesses

Small and medium-sized businesses often have compact teams while maintaining mature, feature-rich products. 
Kotlin Multiplatform allows them to share core logic while keeping the native look-and-feel users expect. 

> By leveraging existing codebases, these teams can accelerate development without compromising on user experience.

Reviewer: ```suggestion
By relying on existing codebases, these teams can accelerate development without compromising on user experience.
```

[Wonder App by Baidu](https://kotlinlang.org/lp/multiplatform/case-studies/baidu)

[![Learn from KMP success stories](kmp-success-stories.svg){width="700"}{style="block"}](case-studies.topic)

### Agencies

Working with diverse clients, agencies and consultancies must accommodate a wide range of platform requirements and business goals. 
The ability to reuse code with Kotlin Multiplatform is especially valuable for teams managing multiple projects under 
tight timelines and limited engineering teams. By adopting KMP, agencies can accelerate delivery and maintain consistent 
app behavior across platforms. 

**Case studies:**

> * [IceRock](https://icerockdev.com/) is an outsourcing company that leverages Kotlin Multiplatform to develop apps for its clients. 

Reviewer: ```suggestion
* [IceRock](https://icerockdev.com/) is an outsourcing company that uses Kotlin Multiplatform to develop apps for its clients. 
```

  Its app portfolio spans various business requirements, complemented by a substantial collection of open-source 
  Kotlin Multiplatform libraries that enhance the Kotlin Multiplatform development process.
* [Mirego](https://kotlinlang.org/lp/multiplatform/case-studies/mirego/), an end-to-end digital product team, uses Kotlin Multiplatform to run the same business logic on the web, 
  iOS, tvOS, Android, and Amazon Fire TV. KMP allows it to streamline development while still getting the most out of each platform.  

### Companies expanding to new markets

Some companies want to enter new markets by launching their apps on platforms they haven't previously targeted, 
for example, moving from iOS-only to include Android or vice versa.

> KMP helps you leverage existing iOS code and development practices while maintaining native performance and UI flexibility on Android. 

Reviewer: ```suggestion
KMP helps you utilize existing iOS code and development practices while maintaining native performance and UI flexibility on Android. 
```

> If you want to maintain platform-specific user experiences and leverage existing knowledge and code, 

Reviewer: ```suggestion
If you want to maintain platform-specific user experiences and take advantage of existing knowledge and code, 
```

### E-commerce

Building cross-platform shopping experiences means balancing shared business logic with native features like payments, 
camera access, and maps. Kotlin Multiplatform with Compose Multiplatform enables teams to share both business logic 
and UIs across platforms, while still using platform-specific components where needed. 
This hybrid approach ensures faster development, a consistent user experience, and the flexibility to integrate critical native features.

**Case studies:** Balary Market, Markaz

### Social networking and community

On social platforms, timely feature delivery and consistent interactions are essential for keeping communities active 
and connected across devices. Key interaction logic might include messaging, notifications, or scheduling. 
For example, Meetup, which allows users to find local groups, events, and activities, has been able to release 

> new features simultaneously by leveraging KMP.

Reviewer: ```suggestion
new features simultaneously thanks to KMP.
```

### Postal services

While not a common use case, Kotlin Multiplatform has even been adopted by a 377-year-old national postal service. 
Norway’s Posten Bring uses KMP to unify complex business logic across dozens of frontend and backend systems, 
helping them streamline workflows and drastically reduce the time required to roll out new services — from months to days.

**Case study:** [Posten Bring](https://2024.javazone.no/program/a1d9aeac-ffc3-4b1d-ba08-a0568f415a02)

These examples highlight how Kotlin Multiplatform can be used in practically any industry or type of app. 
Whether you are building a fintech app, a mobility solution, an education platform, or something else, 
Kotlin Multiplatform provides the flexibility to share as much code as makes sense for your project, 

> without sacrificing the native experience. You can also check out an extensive list of [KMP case studies](case-studies.topic) on its website, 

Reviewer: ```suggestion
without sacrificing the native experience. You can also check out an extensive list of [KMP case studies](case-studies.topic), 
```
